### ISSUE TITLE
Morpheus GraphQL Client with Safe Queries/Mutations

### ISSUE THREAD

**Issue by sordina** [morpheusgraphql/morpheus-graphql#200]
Spinning off from https://github.com/morpheusgraphql/morpheus-graphql/issues/184

* Currently we develop some Rails applications that expose a GraphQL API
* I would like to consume that API with Haskell Applications
* It would be good to have a code-gen tool that takes these schemas and generates Haskell
* There should be a way to perfrom a parameterized query/mutation in 3-phases:
  1. Codegen from the Schema (Haskell implementation of the queries/mutations/types)
  2. Codegen from the query
    - Check that the query matches the schema
    - Generate a Haskell function to perform the query with just the required params/fields
    - Generate a datatype representing the response from that query (possibly with lenses for fields) so that there are no Maybe fields for requested non-null fields.
  3. Run the query by calling the function, validating and marshaling the results.

Let me know if this is a duplicate issue!

**Comment by nalchevanidze** [comment]
idea: 

schema :: IO GQLSchema
schema = ...

```haskell
query schema [ gql |
   query GetHero {
       name
       hobby
       age
   }
]
```
should generate: 

- Data type Declaration `GetHero`:
  ```haskell
    data GetHero = GetHero {
       name :: Text,
       hobby ::  Maybe Text
       age :: Int
    }
 ```

- instance of Typeclass `Fetch` where 

  ```haskell
    class Fetch a where 
      fetch :: (Request -> m Value) -> Either Error a
 ```

- `lens` instance for `GetHero`

**Comment by nalchevanidze** [comment]
included in release `v0.2.1`
